src/main/perl/Path.pm

```diff
@@ -214,17 +216,20 @@ sub _safe_eval
     %opts = %$optsref if $optsref;
 
     local $@;
-    eval {
-        $res = $funcref->(@args, %opts);
-    };
+    my $status = eval {
+                       $res = $funcref->(@args, %opts);
+                      };
 
-    if ($@) {
-        chomp($@);
-        return $self->fail("$failmsg: $@");
-    } else {
-        my $res_txt = defined($res) ? "$res" : '<undef>';
-        chomp($res);
+    # $status is undef if there is a syntax or runtime error or if $res is undef.
+    if ( defined($status) ) {
```

Comment:
you shouldn't use safe_eval on the LC::Check functions. LC::Check throws an error on failure and returns undef, but that's exactly what the `LC_Check` wrapper handles.
safe_eval is for "normal" perl functions.

i don't mind the code; but checking `$@` separately also makes sense, it can catch errors when functions return undef.